docs/integration-docs-guide.md

```diff
@@ -0,0 +1,273 @@
+  integration, including what URLs to use, etc.  If there are any
+  screens in the product involved, take a few screenshots with the
+  input fields filled out with sample values in order to make the
+  instructions really easy to follow.  For the screenshots, use a bot
+  with a name like "GitHub Bot", and an email address for the bot like
+  `github-bot@zulip.example.com`. Zulip's pre-defined Markdown macros
+  can be used for some of these steps. See
+  [Markdown macros](#markdown-macros) for further details.
+
+* Make sure you've added your integration to
+  `zerver/lib/integrations.py`; this results in your integration
+  appearing on the `/integrations` page.
+
+* You'll need to add a SVG graphic
+  of your integration's logo under the
+  `static/images/integrations/logos/<name>.svg`, where `<name>` is the
+  name of the integration, all in lower case; you can usually find them in the
+  product branding or press page. Make sure to optimize the SVG graphic by
+  running `svgo -f path-to-file`.
+
+  If you cannot find a SVG graphic of the logo, please find and include a PNG
+  image of the logo instead.
+
+* Finally, generate a message sent by the integration and take a
+  screenshot of the message to provide an example message in the
+  documentation. If your new integration is a webhook integration,
+  you can generate such a message from your test fixtures
+  using `send_webhook_fixture_message`:
+
+  ```
+  ./manage.py send_webhook_fixture_message \
+       --fixture=zerver/webhooks/pingdom/fixtures/imap_down_to_up.json \
+       '--url=/api/v1/external/pingdom?stream=stream_name&api_key=api_key'
+  ```
+
+  When generating the screenshot of a sample message, give your test
+  bot a nice name like "GitHub Bot", use the project's logo as the
+  bot's avatar, and take the screenshots showing the stream/topic bar
+  for the message, not just the message body.
+
+## Markdown macros
+
+**Macros** are elements in the format of `{!macro.md!}` that insert common
+phrases and steps at the location of the macros. Macros help eliminate
+repeated content in our documentation.
+
+The source for macros is the Markdown files under
+`templates/zerver/help/include` in the
+[main Zulip server repository](https://github.com/zulip/zulip). If you find
+multiple instances of particular content in the documentation, you can
+always create a new macro by adding a new file to that folder.
+
+Here are a few common macros used to document Zulip's integrations:
+
+### `{!create-stream.md!}` macro
+
+* **About:** Recommends that users create a dedicated stream for a
+  given integration. Usually the first step in setting up an integration
+  or webhook.
+
+* **Contents:**
+  See [source][1].
+  **Note:** `{{ integration_display_name }}` is replaced by
+  [Integration.display_name][2] and `{{ recommended_stream_name }}`
+  is replaced by [Integration.stream_name][3].
+
+* **Example usage:**
+  ```
+  {!create-stream.md!}
+  ```
+
+* **Example rendering:**
+  ```
```

Comment:
This seems to render with Python syntax highlighting on the plaintext, which probably is not what you want.